comp/trace/agent/agent.go

```diff
@@ -99,6 +100,13 @@ func newAgent(deps dependencies) Component {
 		tagger:             deps.Tagger,
 		wg:                 sync.WaitGroup{},
 	}
+
+	if err := apiutil.CreateAndSetAuthToken(); err != nil {
+		log.Errorf("could not set auth token: %s", err)
+	} else {
+		ag.Agent.DebugServer.AddRoute("/config", deps.Config.GetCoreConfigHandler())
```

Comment:
Mind adding a comment to explain why we're adding this route here and not in `trace/api` with the other routes?